invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/LocationList/LocationList.js

```diff
@@ -24,16 +30,38 @@ export default class LocationList extends Component {
     return data.hits.map(row => {
       let serialized = formatter.location.toTable(row);
       serialized['Actions'] = (
-        <Button
-          size="small"
-          content={'Edit'}
-          onClick={() => openRecordEditor(locationApi.url, row.location_pid)}
-        />
+        <>
+          <Button
+            icon={'edit'}
+            onClick={() => openRecordEditor(locationApi.url, row.location_pid)}
+            size="small"
+            title={'Edit Record'}
+          />
+          <DeleteRecordModal
+            headerContent={`Are you sure you want to delete the record Location
+              record with ID ${row.location_pid}?`}
+            deleteFunction={() => this._requestDelete(row.location_pid)}
+            refApiUrl={internalLocationApi.url}
+            checkRefs={() =>
+              internalLocationApi.list(`location_pid:${row.location_pid}`)
+            }
+          />
+        </>
       );
       return omit(serialized, ['Created', 'Updated', 'Link']);
     });
   }
 
+  async _requestDelete(locationId) {
+    await this.deleteLocation(locationId);
+    let { hasError } = this.props;
+    if (!hasError) {
+      remove(this.props.data.hits, entry => entry.location_pid === locationId);
+      // NOTE: setState in order to trigger a render
+      this.setState({ state: this.state });
```

Comment:
Another option is to use https://reactjs.org/docs/react-component.html#forceupdate.

Also could there be issues by updating the props directly rather than changing the data in redux so the changes propagate properly to the component?

Reply:
Actually I left the comment there in order to catch the attention and learn something. 

My thinking behind it was mostly the last sentence of the documentation you mentioned.

`Normally you should try to avoid all uses of forceUpdate() and only read from this.props and this.state in render().`

By setting the assigning the state to the state itself it triggers a render without actually changing anything.

For the redux comment I agree, I should pass it there and propagate the removal from the table from the state.

Reply:
According to https://stackoverflow.com/questions/43841930/in-react-what-is-the-difference-between-setstate-and-forceupdate `forceUpdate()` and setting the state to itself yields the same result (except one doesn't call `shouldComponentUpdate`). Then I suppose the React documentation refers to both ways of forcing an update(?).

Either way, it's probably better to handle removal in Redux to prevent any possible inconsistencies.

Reply:
I think that the filtering of the data should be done inside the `deleteLocation` action. https://github.com/inveniosoftware/invenio-app-ils/blob/e077d69fe49899b8b56f5e2c21b88ebaa5e77bff/invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/LocationList/state/actions.js#L40 here you can get the current state and pass the new data in the action payload.

Reply:
I actually moved it in the reducer, re-review please and will adjust it accordingly.